File: src/main/java/me/ag2s/epublib/domain/GuideReference.java
package me.ag2s.epublib.domain;

import java.io.Serializable;
import me.ag2s.epublib.util.StringUtil;

public class GuideReference extends TitledResourceReference implements Serializable {
   private static final long serialVersionUID = -316179702440631834L;
   public static final String COVER = "cover";
   public static String TITLE_PAGE = "title-page";
   public static String TOC = "toc";
   public static String INDEX = "index";
   public static String GLOSSARY = "glossary";
   public static String ACKNOWLEDGEMENTS = "acknowledgements";
   public static String BIBLIOGRAPHY = "bibliography";
   public static String COLOPHON = "colophon";
   public static String COPYRIGHT_PAGE = "copyright-page";
   public static String DEDICATION = "dedication";
   public static String EPIGRAPH = "epigraph";
   public static String FOREWORD = "foreword";
   public static String LOI = "loi";
   public static String LOT = "lot";
   public static String NOTES = "notes";
   public static String PREFACE = "preface";
   public static String TEXT = "text";
   private String type;

   public GuideReference(Resource resource) {
      this(resource, (String)null);
   }

   public GuideReference(Resource resource, String title) {
      super(resource, title);
   }

   public GuideReference(Resource resource, String type, String title) {
      this(resource, type, title, (String)null);
   }

   public GuideReference(Resource resource, String type, String title, String fragmentId) {
      super(resource, title, fragmentId);
      this.type = StringUtil.isNotBlank(type) ? type.toLowerCase() : null;
   }

   public String getType() {
      return this.type;
   }

   public void setType(String type) {
      this.type = type;
   }
}

File: src/main/kotlin/me/ag2s/epublib/domain/GuideReference.kt
package me.ag2s.epublib.domain

import java.io.Serializable
import me.ag2s.epublib.util.StringUtil

class GuideReference : TitledResourceReference, Serializable {
    private var type: String? = null
    constructor(resource: Resource?) : this(resource, null)
    constructor(resource: Resource?, title: String?) : super(resource, title)
    constructor(resource: Resource?, type: String?, title: String?) : this(resource, type, title, null)
    constructor(resource: Resource?, type: String?, title: String?, fragmentId: String?) : super(resource, title, fragmentId) { this.type = if (StringUtil.isNotBlank(type)) type!!.lowercase() else null }
    fun getType(): String? = type
    fun setType(type: String?) { this.type = type }
    companion object {
        private const val serialVersionUID = -316179702440631834L
        const val COVER = "cover"
        @JvmField var TITLE_PAGE = "title-page"; @JvmField var TOC = "toc"; @JvmField var INDEX = "index"; @JvmField var GLOSSARY = "glossary"; @JvmField var ACKNOWLEDGEMENTS = "acknowledgements"; @JvmField var BIBLIOGRAPHY = "bibliography"; @JvmField var COLOPHON = "colophon"; @JvmField var COPYRIGHT_PAGE = "copyright-page"; @JvmField var DEDICATION = "dedication"; @JvmField var EPIGRAPH = "epigraph"; @JvmField var FOREWORD = "foreword"; @JvmField var LOI = "loi"; @JvmField var LOT = "lot"; @JvmField var NOTES = "notes"; @JvmField var PREFACE = "preface"; @JvmField var TEXT = "text"
    }
}
